### PR TITLE
Update EclipseLink 4.0.3 -> 4.0.4

### PR DESCRIPTION
**Pull request by anija-anil**
Updating EclipseLink 4.0.3 -> 4.0.4.
for : https://github.com/OpenLiberty/open-liberty/issues/29275

### CODE REVIEW

**Comment by anija-anil** [comment]
#build

**Comment by tkburroughs** [comment]
Need to run JPA 3.1 tests in full mode...
#build
#spawn.fullfat.buckets = com.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, com.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, com.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, com.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, com.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec20_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec20_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, com.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec20_jpa_3.1_fat

**Review by tkburroughs — CHANGES_REQUESTED** [review]
See io.openliberty.persistence.3.1.thirdparty/build.gradle  There are a couple of message files that should be removed from the `exclude` lists.

If the message changes haven't been contributed back to EclipseLink yet, then that should be done, and the build.gradle file updated to indicate the excludes should be removed in 4.0.5.

Also, since these are excluded, it is possible new messages have been added that have been missed.  At least try to remove them from exclude to see if there are any new or newly changed messages.

**Comment by anija-anil** [comment]
#build
#spawn.fullfat.buckets = com.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, com.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, com.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, com.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, com.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec20_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec20_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, com.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec20_jpa_3.1_fat

File: dev/io.openliberty.persistence.3.1.thirdparty/resources/org/eclipse/persistence/exceptions/i18n/ValidationExceptionResource.nlsprops [hunk 1]
```diff
@@ -639,5 +639,6 @@
 
 7359=No any ASM service available.
 
-7360=Database password was encrypted by deprecated algorithm. Reencrypt it by `passwordUpdate.sh` from eclipselink.zip bundle
+7360=Database password was encrypted by deprecated algorithm.
```
Comment (tkburroughs) [inline dev/io.openliberty.persistence.3.1.thirdparty/resources/org/eclipse/persistence/exceptions/i18n/ValidationExceptionResource.nlsprops]: Please remove this change.... this change will undo the changes requested by ID during the last up date from 4.0.2 to 4.0.3.... see comments in : https://github.com/OpenLiberty/open-liberty/pull/28824

File: dev/io.openliberty.persistence.3.1.thirdparty/resources/org/eclipse/persistence/internal/localization/i18n/LoggingLocalizationResource.nlsprops [hunk 2]
```diff
@@ -169,9 +169,9 @@ elements={0}{1} elements
 
 enable_batch_operations_as_true_not_supported=WLS native CMP setting "enable-batch-operations-as-true" is not directly supported in EclipseLink CMP. See migration doc for details.
 
-encryptor_script_description=This application internally decrypts an old encrypted password that was used by a previous EclipseLink version and encrypts it by the latest algorithm.
+encryptor_script_description=This application internally decrypt old encrypted password used by some previous version EclipseLink and encrypt it by latest algorithm.
```
Comment (tkburroughs) [inline dev/io.openliberty.persistence.3.1.thirdparty/resources/org/eclipse/persistence/internal/localization/i18n/LoggingLocalizationResource.nlsprops]: Please remove this change.... this change will undo the changes requested by ID during the last up date from 4.0.2 to 4.0.3.... see comments in : https://github.com/OpenLiberty/open-liberty/pull/28824

File: dev/io.openliberty.persistence.3.1.thirdparty/build.gradle [hunk 3]
```diff
@@ -55,9 +55,7 @@ task validateNLS {
         fileset(dir: existingResourceDir) {
           include(name: '**/*Resource.nlsprops')
           exclude(name: '**/TraceLocalizationResource.nlsprops')
-          exclude(name: '**/*Resource_*.nlsprops')
```
Comment (tkburroughs) [inline dev/io.openliberty.persistence.3.1.thirdparty/build.gradle:58]: Please keep these excludes, but change the comments to indicate they need to be removed with 4.0.0.5 update

**Review by tkburroughs — CHANGES_REQUESTED** [review]
Please remove the message file changes.... these changes will undo the changes requested by ID during the last up date from 4.0.2 to 4.0.3.... see comments in : https://github.com/OpenLiberty/open-liberty/pull/28824

It appears the changes requested by ID were never contributed back to EclipseLink or did not make it into 4.0.4. Hopefully they can be contributed back to EclipseLink and be included for the 4.0.5 release.  The excludes can then be removed at that time.

**Comment by anija-anil** [comment]
#build

**Comment by tkburroughs** [comment]
#build
#spawn.fullfat.buckets = com.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, com.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, com.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, com.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, com.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, cnreleastescom.ibm.ws.jpa.tests.spec20_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.query_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.oneXone_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.beanvalidation_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.jpaconfig_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.eclipselink_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.container.checkpoint_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.entitymanager_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.oneXmany_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.packaging_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.entity_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, cnlocatescom.ibm.ws.jpa.tests.spec20_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.inheritance_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.manyXmany_jpa_3.1_fat, com.ibm.ws.jpa.tests.eclipselink.query_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.callback_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.relationships.manyXone_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec10.embeddable_jpa_3.1_fat, com.ibm.ws.jpa.tests.spec20_jpa_3.1_fat

**Comment by helyarp** [comment]
@tkburroughs Should this PR have the **ID review not needed** label?

**Comment by tkburroughs** [comment]
@helyarp Correct; I've added "ID review not needed".  There will be no ID changes with this PR.

**Pull request closed without merging**